flytepropeller/pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -203,6 +205,12 @@ func (e *PluginManager) launchResource(ctx context.Context, tCtx pluginsCore.Tas
 
 	key := backoff.ComposeResourceKey(o)
 
+	err = e.schedulerMgr.Mutate(ctx, o)
```

Comment:
I see this is why... So here is my suggestion, how about we handle the scheduler related changes directly to the plugin itself.

For example
MutateForGangScheduling

Also how about preemption etc if we add that in the future?